test/Microsoft.ML.Tests/Scenarios/Api/AspirationalExamples.cs

```diff
@@ -60,5 +60,95 @@ public void FirstExperienceWithML()
                 PetalLength = 5.1f
             });
         }
+
+
+        [Fact]
+        public void SimpleIrisDescisionTrees()
+        {
+            var env = new ConsoleEnvironment();
+            string dataPath = "iris.txt";
+            // Create reader with specific schema.
+            var dataReader = TextLoader.CreateReader(env, ctx => (
+               label: ctx.LoadText(0),
+               sepalWidth: ctx.LoadFloat(1),
+               sepalLength: ctx.LoadFloat(2),
+               petalWidth: ctx.LoadFloat(3),
+               petalLength: ctx.LoadFloat(4)),
+               dataPath);
+
+
+            var pipeline = data.MakeEstimator()
+                .Append(row => (
+                    // Convert string label to key.
+                    label: row.label.Dictionarize(),
+                    // Concatenate all features into a vector.
+                    features: row.sepalWidth.ConcatWith(row.sepalLength, row.petalWidth, row.petalLength)))
```

Comment:
There was likely a separate discussion on this, but why was `ConcatWith` chosen instead of `Concat` ([used in LINQ](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.concat?redirectedfrom=MSDN&view=netframework-4.7.2#System_Linq_Enumerable_Concat__1_System_Collections_Generic_IEnumerable___0__System_Collections_Generic_IEnumerable___0__))? This applies to all the estimators.

Reply:
If you have a vector column `foo`, then `foo.Concat()` can be interpreted as 'concatenate all slots inside `foo`'. `foo.ConcatWith` doesn't allow for this confusion

---
In reply to: [215141225](https://github.com/dotnet/machinelearning/pull/809#discussion_r215141225) [](ancestors = 215141225)

Reply:
Good point. I think given the similarities with LINQ there is a benefit to staying consistent, but agree that removing confusion is better.